com.unity.shadergraph/Editor/Data/Nodes/Channel/SwizzleNode.cs

```diff
@@ -22,151 +24,107 @@ public SwizzleNode()
+                    break;
+            }
+            RemoveSlotsNameNotMatching(new[] { InputSlotId, OutputSlotId });
         }
 
+        //1.Get vector input value
+        //2.Get validated mask input
+        //3.Swizzle: Remap the vector according to mask
         public void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMode)
         {
-            ValidateChannelCount();
             var outputSlotType = FindOutputSlot<MaterialSlot>(OutputSlotId).concreteValueType.ToShaderString();
             var outputName = GetVariableNameForSlot(OutputSlotId);
             var inputValue = GetSlotValue(InputSlotId, generationMode);
             var inputValueType = FindInputSlot<MaterialSlot>(InputSlotId).concreteValueType;
-            if (inputValueType == ConcreteSlotValueType.Vector1)
-                sb.AppendLine(string.Format("{0} {1} = {2};", outputSlotType, outputName, inputValue));
-            else if (generationMode == GenerationMode.ForReals)
-                sb.AppendLine("{0} {1} = {2}.{3}{4}{5}{6};",
-                    outputSlotType,
-                    outputName,
-                    inputValue,
-                    s_ComponentList[m_RedChannel].ToString(CultureInfo.InvariantCulture),
-                    s_ComponentList[m_GreenChannel].ToString(CultureInfo.InvariantCulture),
-                    s_ComponentList[m_BlueChannel].ToString(CultureInfo.InvariantCulture),
-                    s_ComponentList[m_AlphaChannel].ToString(CultureInfo.InvariantCulture));
-            else
-                sb.AppendLine("{0} {1} = {0}({3}[((int){2} >> 0) & 3], {3}[((int){2} >> 2) & 3], {3}[((int){2} >> 4) & 3], {3}[((int){2} >> 6) & 3]);", outputSlotType, outputName, GetVariableNameForNode(), inputValue);
-        }
+            var InputValueSize = SlotValueHelper.GetChannelCount(inputValueType);
 
-        public override void CollectShaderProperties(PropertyCollector properties, GenerationMode generationMode)
-        {
-            base.CollectShaderProperties(properties, generationMode);
-            if (generationMode != GenerationMode.Preview)
-                return;
-            properties.AddShaderProperty(new Vector1ShaderProperty
+            if (!ValidateMaskInput(InputValueSize))
             {
-                overrideReferenceName = GetVariableNameForNode(),
-                generatePropertyBlock = false
-            });
-        }
-
-        public override void CollectPreviewMaterialProperties(List<PreviewProperty> properties)
-        {
-            base.CollectPreviewMaterialProperties(properties);
-            // Encode swizzle values into an integer
-            var value = ((int)redChannel) | ((int)greenChannel << 2) | ((int)blueChannel << 4) | ((int)alphaChannel << 6);
-            properties.Add(new PreviewProperty(PropertyType.Float)
+                owner.AddValidationError(objectId, "Invalid mask!", ShaderCompilerMessageSeverity.Error);
```

Comment:
No exclamation points in error messages please. It's not exciting =). Can we give them a hint what would be valid instead? (Like, "Invalid Mask. Valid values are r, g, b.")

Reply:
Took off the exclamation mark. And for the error message what do you think about this?
![Screenshot (4)](https://user-images.githubusercontent.com/67800322/102411763-513d6b00-3fa7-11eb-9cb6-ecf7b8bc8cd3.png)